### ISSUE TITLE
local git repository in openshift 3

### ISSUE THREAD

**Issue by yeganx** [openshift/origin#13689]
we have problem in build source code that is local git rep

##### Version
```
openshift origin
Version
oc v1.4.1
kubernetes v1.4.0+776c994
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://localhost:8443
openshift v1.4.1
kubernetes v1.4.0+776c994
```

##### Steps To Reproduce
all my openshift node can clone it , but its not password less
```
 $ git clone user@192.168.0.11:/home/user/gitRepSrc
```

my type of project is php project

i go through this https://blog.openshift.com/deploying-from-private-git-repositories/

in 192.168.0.11 I did
```
$ oc project test
$ ssh-keygen -t rsa -C "my_secret_key_for_OpenShift"
Enter file in which to save the key (/root/.ssh/id_rsa): /root/.ssh/id_xxx
Enter passphrase (empty for no pass
Enter same passphrase again: (enter)
$ oc secrets new-sshauth sshsecret --ssh-privatekey=$HOME/.ssh/id_XXX
$ oc secrets add serviceaccount/builder secrets/sshsecret
$ oc patch buildConfig myapp -p '{"spec":{"source":{"sourceSecret":{"name":"sshsecret"}}}}'
```

##### Current Result
errors in build log
```
error: build error: Warning: Permanently added '192.168.0.11' (ECDSA) to the list of known hosts.
Permission denied, please try again.
Permission denied, please try again.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
```

**Comment by bparees** [comment]
does your key have a passphrase?  openshift does not support keys w/ passphrases since there's no interactive point in the build for it to prompt the user for the passphrase.

otherwise can you provide the full buildconfig json?

you may also find the answer to your problem by reviewing our docs w/ respect to using secrets during git cloning, if you have not already: https://docs.openshift.org/latest/dev_guide/builds/build_inputs.html#source-clone-secrets

**Comment by yeganx** [comment]
It,s my build config yaml file
apiVersion: v1
kind: BuildConfig
metadata:
  name: test7
  namespace: java
  selfLink: /oapi/v1/namespaces/java/buildconfigs/test7
  uid: 1f8b1ddb-1da3-11e7-ac45-fa163eda6df3
  resourceVersion: '3337051'
  creationTimestamp: '2017-04-10T04:06:50Z'
  labels:
    app: test7
  annotations:
    openshift.io/generated-by: OpenShiftWebConsole
spec:
  triggers:
    - type: Generic
      generic:
        secret: 47ff8aebbf5b36fe
    - type: GitHub
      github:
        secret: 0f951ec5bd7df20c
    - type: ImageChange
      imageChange:
        lastTriggeredImageID: 'x.x.x.x:5000/jboss-eap-7/eap70-openshift:1.4'
    - type: ConfigChange
  runPolicy: Serial
  source:
    type: Git
    git:
      uri: 'user@192.168.0.11:/home/user/gitRepSrc'
      ref: master
    sourceSecret:
      name: basicsec
  strategy:
    type: Source
    sourceStrategy:
      from:
        kind: ImageStreamTag
        namespace: openshift
        name: 'jboss-eap70-openshift:1.4'
  output:
    to:
      kind: ImageStreamTag
      name: 'test7:latest'
  resources: {}
  postCommit: {}
  nodeSelector: null
status:
  lastVersion: 5

secrete I created :+1: 
oc secrets new-basicauth basicsec \
    --username=<user_name> \
    --password=<password> \
    --gitconfig=/home/user/gitRepSrc/.git/config

my ./git/config file is :+1: 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url =  user@192.168.0.11:/home/user/gitRepSrc
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[http]
        sslVerify=false

eeror in build 
Cloning "user@192.168.0.11:/home/user/gitRepSrc" ...
error: build error: Host key verification failed.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

**Comment by bparees** [comment]
`error: build error: Host key verification failed.`

this points to a TLS key verification problem, see:
https://docs.openshift.org/latest/dev_guide/builds/build_inputs.html#source-secrets-trusted-certificate-authorities

**Comment by bparees** [comment]
it sounds like you're having the same issue that i just helped another user w/ here:
https://github.com/openshift/origin/issues/13863

i'm going to close this, but if you still have not resolved your issue and reading through #13863 does not help (in particular the last few comments where we zeroed in on the configuration issue), come back and comment here.
